backend/substrapp/tasks/tasks.py

```diff
@@ -916,19 +919,43 @@ def remove_algo_images(algo_hashes):
     client = docker.from_env()
     for algo_hash in algo_hashes:
         algo_docker = get_algo_image_name(algo_hash)
-        logger.info(f'Remove docker image {algo_docker}')
-        client.images.remove(algo_docker, force=True)
+
+        try:
+            if client.images.get(algo_docker):
+                client.images.remove(algo_docker, force=True)
+                logger.info(f'Remove docker image {algo_docker}')
+
+        except docker.errors.ImageNotFound:
+            pass
+        except docker.errors.APIError as e:
+            logger.exception(e)
+
+
+def remove_intermediary_models(model_hashes):
+    from substrapp.models import Model
+
+    models = Model.objects.filter(pkhash__in=model_hashes, validated=True)
+    filtered_model_hashes = [model.pk for model in models]
+
+    models.delete()
```

Comment:
aren't the models also stored on disk?

Reply:
Yes. Models are stored on the disk :-)
But there is a model signal that delete it when the instance is deleted !

Reply:
Would you mind pointing me to the piece of code that does that?

Also: have you checked the files actually get deleted on disk?

Reply:
You can find the code here : https://github.com/SubstraFoundation/substra-backend/blob/master/backend/substrapp/signals/model/post_delete.py
Linked here : https://github.com/SubstraFoundation/substra-backend/blob/master/backend/substrapp/apps.py#L26
Yes, I've checked that models are removed from db and deleted on disk

Reply:
Thnks for the links!